File: FFmpegExamples/avio_reading/KSMediaPlayer/ViewController.m
//
//  ViewController.m
//  KSMediaPlayer
//
//  Created by saeipi on 2020/6/1.
//  Copyright © 2020 saeipi. All rights reserved.
//  

#import "ViewController.h"
#include "avio_reading.h"
@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSButton *playBtn = [NSButton buttonWithTitle:@"播放" target:self action:@selector(mediaPlayer)];
    [self.view addSubview:playBtn];
    playBtn.frame = CGRectMake(self.view.frame.size.width / 2 - 50, self.view.frame.size.height / 2 - 30, 100, 60);
    
}

- (void)mediaPlayer {
    char *url = "/Users/saeipi/Downloads/File/SOPSandwich.mp4";
    
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}


@end

File: KSEncodeAAC/KSEncodeAAC/ViewController.swift
//
//  ViewController.swift
//  KSAudioResample
//
//  Created by saeipi on 2020/5/18.
//  Copyright © 2020 saeipi. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    var recStatus: Bool = false;
    var thread: Thread?
    var btnAction: NSButton?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.view.setFrameSize(NSSize(width: 320, height: 240))
        btnAction = NSButton.init(title: "开始录制", target: self, action: #selector(buttonClick))
        btnAction?.bezelStyle = .rounded
        btnAction?.setButtonType(.pushOnPushOff)
        btnAction?.frame = NSRect(x: 320/2-50, y: 240/2-15, width: 100, height: 30)
        self.view.addSubview(btnAction!)
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    
    @objc func buttonClick() {
        self.recStatus = !self.recStatus
        if self.recStatus {
            thread = Thread.init(target: self, selector: #selector(recordAudio), object: nil);
            thread?.start()
            btnAction?.title = "停止录制"
        }
        else{
            update_status(0)
            btnAction?.title = "开始录制"
        }
    }
    
    @objc func recordAudio() {
        record_audio()
    }

}

